src/stores/RolleStore.ts

```diff
@@ -63,6 +66,30 @@ export const useRolleStore: StoreDefinition<'rolleStore', RolleState, RolleGette
     };
   },
   actions: {
+    async addServiceProviderToRolle(rolleId: string, rolleServiceProviderQueryParams: RolleServiceProviderQueryParams) {
+      this.loading = true;
+      try {
+        const { data }: AxiosResponse<ServiceProviderResponse> = await rolleApi.rolleControllerAddServiceProviderById(
+          rolleId,
+          rolleServiceProviderQueryParams,
+        );
+        if (this.createdRolle) {
+          if (this.createdRolle.serviceProviders) {
```

Comment:
Besser so vielelicht?

```suggestion
       if (this.createdRolle?.serviceProviders) {
```

Reply:
Das hatte ich erst so, aber dann meckert TS im else bei der Zuweisung, dass createdRolle undefined sein könnte und wenn ich dann `createdRolle?.serviceProviders` schreibe, kommt `The left-hand side of an assignment expression may not be an optional property access`. Wenn wir das lösen könnten, würde ich auch das optional chaining im if benutzen.

Reply:
Oh ja... Dann vielleicht geht das auch so?

```suggestion
if (this.createdRolle && this.createdRolle.serviceProviders) {
  // Your code here
}
```

Reply:
Oh ja... dann vielleicht geht das so besser, dann haben wir keine nested Ifs

```suggestion
     if (this.createdRolle?.serviceProviders) {
}
```

Reply:
Das geht auch nicht, weil der else Teil dann nicht mit drin ist und das gleiche Problem auftritt.

Reply:
Das Ursprungsproblem ist, dass die Prop serviceProviders nicht von Anfang an in der createdRolle verfügbar ist und man keine Prop erstellen kann, wenn man reinpusht. Kennst du da vllt noch einen Trick?

Reply:
```suggestion
        if (this.createdRolle) {
          this.createdRolle.serviceProviders = this.createdRolle.serviceProviders || [];
          this.createdRolle.serviceProviders.push(data);
        }
```
Hier wenn createdRolle.serviceProviders null ist oder undefined dann wird es als Empty array initialisiert und data wird dahin gepushed.

Reply:
Nehm ich, danke!